File: FilterWordsStartingWithVowel.py
# Filter Words Starting With Vowel
# - Given a list of words, return only those that start with a vowel.
# - Time: O(n), Space: O(n)

class Solution:
    def filterStartingWithVowel(self, words: list[str]) -> list[str]:
        vowels = set("aeiouAEIOU")
        return [w for w in words if w and w[0] in vowels]

File: FilterWordsStartingWithVowel.cs
// Filter Words Starting With Vowel
// - Given a list of words, return only those that start with a vowel.
// - Time: O(n), Space: O(n)

using System;
using System.Collections.Generic;
using System.Linq;

class Solution
{
    public List<string> FilterStartingWithVowel(List<string> words)
    {
        HashSet<char> vowels = new HashSet<char> { 'a', 'e', 'i', 'o', 'u', 
                                                   'A', 'E', 'I', 'O', 'U' };

        return words.Where(w => !string.IsNullOrEmpty(w) && vowels.Contains(w[0])).ToList();
    }
}
